### PR TITLE
Add tests to gtest-parallel.

### PR DESCRIPTION
**Pull request by ehlemur-zz**

### CODE REVIEW

**Comment by ehlemur-zz** [comment]
@pbos @kwiberg 
PTAL.

**Comment by pbos** [comment]
Can you add these tests one at a time so they're easier to review? :)

**Comment by ehlemur-zz** [comment]
Ok :)
PTAL

File: gtest_parallel.py [hunk 1]
```diff
@@ -177,11 +177,12 @@ class TaskManager(object):
   Logger, TestResults and TestTimes classes, and in case of failure, retries the
   test as specified by the --retry_failed flag.
   """
-  def __init__(self, times, logger, test_results, times_to_retry,
+  def __init__(self, times, logger, test_results, task_factory, times_to_retry,
```
Comment (pbos) [inline gtest_parallel.py:180]: Is this a factory, or should this be called task_class? Python kinda makes this the same thing. WDYT?

File: gtest_parallel_tests.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
+import collections
```
Comment (pbos) [inline gtest_parallel_tests.py:15]: Add license header w/ 2017.

File: gtest_parallel.py (hunk 1 above)
Comment (pbos) [reply gtest_parallel.py:180]: Ignore this, I think it's fine. :)

**Review by kwiberg — CHANGES_REQUESTED** [review]
Looks good, except the copyright year as pbos@ said.

**Comment by ehlemur-zz** [comment]
PTAL :)

**Pull request merged**